pkg/node/manager/manager.go

```diff
@@ -292,13 +308,25 @@ func (m *manager) backgroundSync(ctx context.Context) error {
 			entry.mutex.Lock()
 			m.mutex.RUnlock()
 			m.Iter(func(nh datapath.NodeHandler) {
-				nh.NodeValidateImplementation(entry.node)
+				if err := nh.NodeValidateImplementation(entry.node); err != nil {
+					log.WithError(err).Errorf("Failed to validate handler %s on node %s", nh.Name(), entry.node.Name)
+					errs = errors.Join(errs, fmt.Errorf("handling %s on node %s: %w", nh.Name(), entry.node.Name, err))
+				}
 			})
 			entry.mutex.Unlock()
 
 			m.metrics.DatapathValidations.Inc()
 		}
 
+		if errs != nil {
+			log.WithError(errs).Debug("Failed to do node validation")
+			m.healthReporter.Degraded("Failed to do node validation", errs)
```

Comment:
Do we need both debug logs and health reporter logs? Maybe we can put a debug log inside the health reporter instead of calling `.Debug` at each call site of the health reporter.

Reply:
Probably not actually, the health reporter has debug logging on emitting status updates (which we may consider making into info at some point).